src/components/employee/addEmployeeModal.js

```diff
@@ -0,0 +1,259 @@
+import { useContext, useState, useEffect } from 'react'
+import { EmployeeContext } from '../../contexts/employeeContext'
+import { TechnicalContext } from '../../contexts/technicalContext'
+import { ComponentsContext } from '../../contexts/componentsContext'
+
+import { Form, Upload, message, Modal } from 'antd';
+import { LoadingOutlined, PlusOutlined } from '@ant-design/icons';
+import TextInput from '../inputs/InputTextCommon'
+import Checkbox from '../inputs/CheckBoxCommon'
+import RadioButton from '../inputs/RadioCommon'
+import Button from '../buttons/ButtonCommon'
+
+const AddEmployeePage = () => {
+    const {
+        technicalState: { technicals },
+        getTechnicals
+    } = useContext(TechnicalContext);
+
+    const {
+        createEmployee,
+        showModal,
+        setShowModal
+    } = useContext(EmployeeContext);
+
+    const {
+        checkedItems,
+        radioItem
+    } = useContext(ComponentsContext);
+
+    useEffect(() => { getTechnicals() }, [])
+
+    let techOptions = []
+    technicals.map(tech => (
+        techOptions.push(
+            {
+                label: tech.name,
+                value: tech._id,
+            }
+        )
+    ))
+
+    const genderOptions = [
+        {
+            label: 'Male',
+            value: 'male',
+        },
+        {
+            label: 'Female',
+            value: 'female',
+        },
+        {
+            label: 'Other',
+            value: 'other',
+        },
+    ]
+
+    const onFinish = (values) => {
+        console.log(values)
+        const formData = new FormData()
+        formData.append("image", imgFile);
+        formData.append("name", form.getFieldValue("name"));
+        formData.append("code", form.getFieldValue("code"));
+        formData.append("phone", "+84" + form.getFieldValue("phone"));
+        formData.append("email", form.getFieldValue("email"));
+        formData.append("identity", form.getFieldValue("identity"));
+        formData.append("gender", radioItem);
+        formData.append("technical", JSON.stringify(checkedItems));
+
+        createEmployee(formData)
+        setShowModal(false);
+    };
+
+    const onFinishFailed = (errorInfo) => {
+        console.log('Failed:', errorInfo);
+        console.log(form.getFieldValue("name"))
+    };
+
+    const [loading, setLoading] = useState(false);
+    const [imageUrl, setImageUrl] = useState();
+    const [imgFile, setImgFile] = useState();
+
+    const getBase64 = (img, callback) => {
+        const reader = new FileReader();
+        reader.addEventListener('load', () => callback(reader.result));
+        reader.readAsDataURL(img);
+    };
+
+    const beforeUpload = (file) => {
+        const isJpgOrPng = file.type === 'image/jpeg' || file.type === 'image/png';
+        if (!isJpgOrPng) {
+            message.error('You can only upload JPG/PNG file!');
```

Comment:
khi check điều kiện. Nên dùng Return để ngưng việc tiếp tục thực thi function. Tránh trường hợp mặc dù notification thông báo lỗi nhưng những tác vụ phía sau vẫn tiếp tục thực thi:

return message.error('You can only upload JPG/PNG file!');